### PR TITLE
[DEV-814] fiat amounts for withdrawals and deposits

### PR DESCRIPTION
**Pull request by TonySemikin**

### CODE REVIEW

File: src/subdomains/staking/domain/entities/staking.entity.ts [hunk 1]
```diff
@@ -232,6 +253,31 @@ export class Staking extends IEntity {
     return this.getWithdrawalsByStatus(WithdrawalStatus.PAYING_OUT);
   }
 
+  getWithdrawalsWithoutFiatReferences(): Withdrawal[] {
+    return this.withdrawals.filter(
+      (w) =>
+        (!w.amountChf && w.amountChf !== 0) ||
```
Comment (davidleomay) [inline src/subdomains/staking/domain/entities/staking.entity.ts:259]: Why not just `w.amountChf == null`?

File: src/subdomains/staking/application/services/staking.service.ts [hunk 2]
```diff
@@ -149,4 +179,96 @@ export class StakingService {
...
+  }
+
+  private removeStakingReferencesDuplicates(stakings: StakingReference[] = []): StakingReference[] {
+    return stakings.reduce((res, curr) => {
+      const existing = res.find((r) => r.stakingId === curr.stakingId);
+
+      !existing && res.push(curr);
+
+      return res;
+    }, []);
```
Comment (davidleomay) [inline src/subdomains/staking/application/services/staking.service.ts:272]: It should be possible to remove duplicates using
```ts
stakings.filter((item, pos, self) => self.indexOf(item) === pos);
```

File: src/subdomains/staking/application/services/staking.service.ts [hunk 3]
```diff
@@ -149,4 +179,96 @@ export class StakingService {
...
+
+    // not querying Stakings, because eager query is not supported, thus unsafe to fetch entire entity
+    const queryByWithdrawals = await this.repository
+      .createQueryBuilder('staking')
+      .innerJoin('staking.withdrawals', 'withdrawal')
+      .where('withdrawal.status = :withdrawalStatus', { withdrawalStatus: WithdrawalStatus.CONFIRMED })
+      .andWhere('(withdrawal.amountEur IS NULL OR withdrawal.amountUsd IS NULL OR withdrawal.amountChf IS NULL)')
+      .leftJoinAndSelect('staking.asset', 'asset')
+      .getMany()
+      .then((s) => s.map((i) => ({ stakingId: i.id, assetId: i.asset.id })));
```
Comment (davidleomay) [inline src/subdomains/staking/application/services/staking.service.ts:202]: Why separated queries?

File: src/subdomains/staking/application/services/staking.service.ts (hunk 2 above)
Comment (TonySemikin) [reply src/subdomains/staking/application/services/staking.service.ts]: addressed

File: src/subdomains/staking/domain/entities/staking.entity.ts (hunk 1 above)
Comment (TonySemikin) [reply src/subdomains/staking/domain/entities/staking.entity.ts]: addressed

File: src/subdomains/staking/application/services/staking.service.ts (hunk 3 above)
Comment (TonySemikin) [reply src/subdomains/staking/application/services/staking.service.ts]: addressed

**Review by davidleomay — APPROVED** [review]

**Review by davidleomay — APPROVED** [review]

**Pull request merged**